io/log_stream.go

```diff
@@ -0,0 +1,38 @@
+package io
+
+import (
+	"context"
+	"io"
+)
+
+type logBody struct {
+	ctx    context.Context
```

Comment:
Is this used?

Reply:
no, but maybe we should stop processing if it's canceled, wdyt?

Reply:
I changed to inspect ctx.Done() before reading

Reply:
added testcase to test cancelation

Reply:
Why do we care about context here? we don't care elsewhere and I think we should not use it here for simplicity. Also for response path its unclear which context to use so one has to resort for context.Background, i.e. its unused

Reply:
We can use the request.Context() form FilterContext, if that's canceled as in client canceled case we can stop processing.